### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by Plateria**
Added tests and removed toString() from subject.

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
```
Comment (stigito) [inline pubspec.lock]: as mentioned dont commit this file ^^

File: test/unit_tests/database_test.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
+void main() {
```
Comment (stigito) [inline test/unit_tests/database_test.dart:1]: dead code

Comment (stigito) [reply test/unit_tests/database_test.dart:1]: example for later db testing

File: test/unit_tests/exam_type_test.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    var result = ExamType.getType("Oral exam");
+    var compare = ExamType.ORAL_EXAM;
+    expect(result, compare);
+  });
+
+  test('exam type getType Written exam', () {
+    var result = ExamType.getType("Written exam");
+    var compare = ExamType.WRITTEN_EXAM;
+    expect(result, compare);
+  });
```
Comment (stigito) [inline test/unit_tests/exam_type_test.dart:28]: maybe add a test which tries to get an empty text. because this was the case because our creating sometimes messes up the display list -> endless spinner

Comment (stigito) [reply test/unit_tests/exam_type_test.dart:28]: also with priorities

File: test/unit_tests/priority_test.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// This is a basic Flutter widget test.
```
Comment (stigito) [inline test/unit_tests/priority_test.dart:1]: maybe remove those general comments at the top of each file?

File: test/unit_tests/subject_test.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import 'package:easy_study/model/subject.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  Subject _getDummySubject() {
+    return Subject.name("Software Engineering II", ExamType.WRITTEN_EXAM,
+        "T1.011", Priority.MINIMALISM, "A funny subject.", 5);
+  }
+
+  test('subject to map', () {
```
Comment (stigito) [inline test/unit_tests/subject_test.dart:19]: you should rebase or create an issue where you add the attributes "time left" and "startedTimetrackingAt" to the test of toMap()

File: test/widget_tests/screens/screens_test.dart [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    await tester.pump();
+  });
+
+  testWidgets('show home', (tester) async {
+    var createApp = _createApp(Home());
+    await tester.pumpWidget(createApp);
+    await tester.pump();
+  });
+
+  testWidgets('show setings', (tester) async {
```
Comment (stigito) [inline test/widget_tests/screens/screens_test.dart]: spelling

File: test/widget_tests/screens/screens_test.dart [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    return app;
+  }
+
+  testWidgets('show subject overview', (tester) async {
+    var createApp = _createApp(SubjectOverview());
+    await tester.pumpWidget(createApp);
+    await tester.pump();
+  });
+
+  testWidgets('show subject card', (tester) async {
```
Comment (stigito) [inline test/widget_tests/screens/screens_test.dart:40]: if you add the card test you should rebase or create an issue to add a "subjectprogressbar" test

**Review by stigito — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 1 above)
Comment (Plateria) [reply pubspec.lock]: we will do this later

File: test/unit_tests/database_test.dart (hunk 2 above)
Comment (Plateria) [reply test/unit_tests/database_test.dart:1]: as u mentioned, we will add this feature later

**Review by stigito — APPROVED** [review]
